enhancements/machine-config/manage-boot-images.md

```diff
@@ -0,0 +1,197 @@
+---
+title: manage-boot-images
+authors:
+  - "@djoshy"
+reviewers: 
+  - "@yuqi-zhang"
+  - "@mrunal"
+  - "@cgwalters, for rhcos context" 
+  - "@joelspeed, for machine-api context" 
+  - "@sdodson, for installer context"
+approvers:
+  - "@yuqi-zhang"
+api-approvers: 
+  - "@joelspeed" 
+  - "@murnal" 
+creation-date: 2023-10-05
+last-updated: 2022-10-05
+tracking-link:
+  - https://issues.redhat.com/browse/MCO-589
+see-also:
+replaces:
+superseded-by: https://github.com/openshift/enhancements/pull/201, https://github.com/openshift/enhancements/pull/368
+---
+
+# Managing boot images via the MCO
+
+## Summary
+
+This is a proposal to manage bootimages via the `Machine Config Operator`(MCO), leveraging some of the [pre-work](https://github.com/openshift/installer/pull/4760) done as a result of the discussion in [#201](https://github.com/openshift/enhancements/pull/201). 
+
+For Install Provisioned Infrastructure(IPI) clusters, the end goal is to create a mechanism that can:
+- update the boot images references in `MachineSets` to the latest in the payload image
```

Comment:
Note that this is different in standalone vs hypershift - the latter uses cluster API. 

AIUI, hypershift already uses the referenced bootimage metadata?  Yeah, looks like it; see e.g.
https://github.com/openshift/hypershift/blob/32309b12ae6c5d4952357f4ad17519cf2424805a/hypershift-operator/controllers/nodepool/nodepool_controller.go#L2168

So I think this patching already happens when updating node pools.

We could probably hence rephrase this to mainly talk about "standalone" OCP installs?

But I think there is a detail here to make sure that hypershift cases also are upgraded in a similar/uniform way where applicable.

Reply:
Understood, I'll make sure to call this out!